src/api/postApi.js

```diff
@@ -0,0 +1,63 @@
+import axios from "axios";
+import { useSelector } from "react-redux";
+import IP_ADDRESS from './IPAddress';
+
+const PostApi = () => {
+  const currentUserId = useSelector((state) => {
+    return state.userDataReducer[0]?.data?.user.id;
+  });
+
+  const createPost = async (datapost) => {
+    let data = JSON.stringify({
+      userId: currentUserId,
+      body: datapost.postDesc,
+      attachment: datapost.postMedia,
+    });
+
+    let config = {
+      method: "post",
+      maxBodyLength: Infinity,
+      url: `${IP_ADDRESS}/v1/post/`,
+      headers: {
+        token: localStorage.getItem("token"),
+        "Content-Type": "application/json",
+      },
+      data: data,
+    };
+
+    const response = axios.request(config);
+    return response;
+  };
+
+  const getPosts = async () => {
+    let config = {
```

Comment:
As you can see the basic structure of config in each and every API call is similar, just the API method, and end point is different, so create a separate function called Request(method, endPoint) {config and api call}; and use this Request method for all API calls.